asdf/_asdf.py

```diff
@@ -182,9 +179,13 @@ def __init__(
         self._closed = False
         self._external_asdf_by_uri = {}
         # if memmap is set, it overrides copy_arrays
-        if memmap is not NotSet:
-            copy_arrays = not memmap
-        self._blocks = BlockManager(uri=uri, lazy_load=lazy_load, memmap=not copy_arrays)
+        if copy_arrays is not NotSet:
+            warnings.warn("copy_arrays is deprecated; use memmap instead", AsdfDeprecationWarning)
```

Comment:
Let's use a non-`DeprecationWarning` here (maybe just `AsdfWarning`). Python by default will hide deprecation warnings not originating in `__main__` and we want all users to see this (even those using jupyter etc).

Also let's expand the message to include something about the upcoming `memmap=False` default change. This is pretty much the only opportunity we have to communicate this change to users. Maybe something like:
```
copy_arrays is deprecated; use memmap instead. Note that memmap will default to False in asdf 4.0.
```